checkerboard_detector/src/checkerboard_detector.cpp

```diff
@@ -95,11 +97,19 @@ class CheckerboardDetector
     map<string,int> maptypes;
     ros::Time lasttime;
     boost::mutex mutexcalib;
+    boost::mutex mutexconfig;
```

Comment:
I suggest to delete `mutexconfig` and rename `mutexcalib` to `mutex` (for more general use), and use it also for locking callback for configure.